include/nlohmann/detail/json_custom_base_class.hpp

```diff
@@ -0,0 +1,30 @@
+#pragma once
+
+#include <type_traits> // conditional, is_same
+
+namespace nlohmann
+{
```

Comment:
Please use `NLOHMANN_JSON_NAMESPACE_BEGIN` to open the `nlohmann` namespace here and `NLOHMANN_JSON_NAMESPACE_END` at the end of the file to close it.

```suggestion
NLOHMANN_JSON_NAMESPACE_BEGIN
```